pkg/providers/amifamily/ami.go

```diff
@@ -94,45 +98,91 @@ func (p *Provider) KubeServerVersion(ctx context.Context) (string, error) {
-				if err := instanceType.Requirements.Compatible(amiRequirements[ami]); err == nil {
+
+	for _, instanceType := range instanceTypes {
+		for _, ami := range amis {
+			if !resources.IsZero(instanceType.Capacity[v1alpha1.ResourceAWSNeuron]) {
+				if err := ami.Requirements.Compatible(
+					scheduling.NewRequirements(
+						scheduling.NewRequirement(v1alpha1.LabelInstanceGPUManufacturer, v1.NodeSelectorOpIn, v1alpha1.NVIDIAGPU, v1alpha1.AWSNeuron)),
+				); err == nil {
 					amiIDs[ami.AmiID] = append(amiIDs[ami.AmiID], instanceType)
 					break
 				}
+			} else if err := instanceType.Requirements.Compatible(ami.Requirements); err == nil {
+				amiIDs[ami.AmiID] = append(amiIDs[ami.AmiID], instanceType)
+				break
 			}
+
 		}
-		if len(amiIDs) == 0 {
-			return nil, fmt.Errorf("no instance types satisfy requirements of amis %v,", lo.Keys(amiRequirements))
+	}
+	return amiIDs
+}
+
+// Returning a list of AMIs with its associated requirements
+// the AMIs are sorted by creation date in descending order.
+// If creation date is nil, the AMIs will be sorted by name in ascending order.
+func (p *Provider) Get(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate, options *Options) ([]AMI, error) {
+	var err error
+	var amis []AMI
+	if len(nodeTemplate.Spec.AMISelector) == 0 {
+		amis, err = p.getDefaultAMIFromSSM(ctx, nodeTemplate, options)
+		if err != nil {
+			return nil, err
 		}
 	} else {
-		for _, instanceType := range instanceTypes {
-			amiID, err := p.getDefaultAMIFromSSM(ctx, amiFamily.SSMAlias(kubernetesVersion, instanceType))
-			if err != nil {
-				return nil, err
-			}
-			amiIDs[amiID] = append(amiIDs[amiID], instanceType)
+		amis, err = p.getAMIsFromSelector(ctx, nodeTemplate)
+		if err != nil {
+			return nil, err
+		}
+
+	}
+	amis = sortAMIsByCreationDate(amis)
+	if len(nodeTemplate.Spec.AMISelector) != 0 {
+		amis = groupAMIsByRequirements(amis)
+	}
+
+	return amis, nil
+}
+
+// Getting the most recent AMIs, by creation date, that have a unique set of requirements
+func groupAMIsByRequirements(amis []AMI) []AMI {
+	var result []AMI
+	requirementsHash := sets.New[uint64]()
+	for _, ami := range amis {
+		hash := lo.Must(hashstructure.Hash(ami.Requirements.NodeSelectorRequirements(), hashstructure.FormatV2, &hashstructure.HashOptions{SlicesAsSets: true}))
+		if !requirementsHash.Has(hash) {
+			result = append(result, ami)
 		}
+		requirementsHash.Insert(hash)
 	}
-	return amiIDs, nil
+	return result
 }
 
-func (p *Provider) getDefaultAMIFromSSM(ctx context.Context, ssmQuery string) (string, error) {
+func (p *Provider) getDefaultAMIFromSSM(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate, options *Options) ([]AMI, error) {
+	amiFamily := GetAMIFamily(nodeTemplate.Spec.AMIFamily, options)
+	kubernetesVersion, err := p.KubeServerVersion(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("getting kubernetes version %w", err)
+	}
+
+	var amis []AMI
+	ssmRequirements := amiFamily.DefaultAMIs(kubernetesVersion)
```

Comment:
The default AMI changes seems like we had a bug previously with our AMI Selection logic. If it is a bug, can we cut a separate PR for this as well?